source/ncengine/graphics2/diligent/resource/GlobalResourceSignature.h

```diff
@@ -10,15 +10,18 @@
 namespace nc::graphics
 {
 class GlobalTextureBufferResource;
+class GlobalEnvironmentResource;
 
 class GlobalResourceSignature
 {
     public:
         static constexpr auto BindingIndex = 0u;
         static constexpr auto SignatureName = "GlobalResourceSignature";
         static constexpr auto GlobalTextureBufferShaderVariableName = "g_textures";
+        static constexpr auto GlobalEnvironmentShaderVariableName = "EnvironmentData";
```

Comment:
Also, I'm not sure what the best way to manage these variables is. Just spitballing, but maybe we could keep them in their own own file, and inject them into shaders with macros and include them here. There would at least be a single source of truth that way.